providers/dns/hetzner/internal/client.go

```diff
@@ -144,17 +146,37 @@ func (c *Client) GetZoneID(domain string) (string, error) {
 }
 
 // https://dns.hetzner.com/api-docs#operation/GetZones
-func (c *Client) getZones() (*Zones, error) {
+func (c *Client) getZones(name string, pagination *Pagination) (*Zones, error) {
```

Comment:
It doesn't look like `getZones` is used outside of `GetZoneID`, hence `pagination` is always `nil`. Do we want to drop it (it is basically dead code now)?

Reply:
I want to keep it for now, because I don't trust API documentation of DNS providers, so I want to wait for the feedback of @MaxKlaxxMiner

Reply:
Sure, sounds good.

Reply:
Ok, all versions of the fix currently work without issues.

I checked the name parameter more closely and found the following:
- If name parameter is sent empty or is omitted completely, then getZones always returns all domains. (as expected)
- If you search with name, the search is always Exact: If you search e.g. "vacando.co" (columbia), also only this one domain is returned and not the "vacando.co.uk" (united kingdom). If you search only "vacando", nothing will be found.

Searching by name can only return a single domain or a 404-error: Maybe a special single-request-function "getZoneByName" without pagnation would be enough here?

Reply:
thank you for the feedback I will remove the pagination paramter

Reply:
Yep and the request by name is also surprisingly much faster:
950-1120 ms (all = 2 pages = 130 domains)
580-660 ms (only 1 page = first 100 domains)
130-150 ms (search per name)